parser.cc

```diff
@@ -1067,7 +1067,9 @@ class TraceExpr : public Expr
 		                con.position.col);
 	}
 public:
+#ifdef DEBUG_PARSING
```

Comment:
This constructor is only called from `trace_debug`, which is only declared in `parser.hh` and defined in `parser.cc` when `DEBUG_PARSING` is defined. I suppose the alternative would be to declare `trace_debug` unconditionally in `parser.hh`, but that would break the inlining optimization that the header seems designed for.